packages/components/src/Checkbox/Checkbox.test.tsx

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import { cleanup, fireEvent, render } from "@testing-library/react";
+import { Checkbox } from ".";
+
+afterEach(cleanup);
+
+it("renders a Checkbox", () => {
+  const tree = renderer
+    .create(<Checkbox label="Send me spam?" name="send_me_span" value="spam" />)
+    .toJSON();
+  expect(tree).toMatchSnapshot();
+});
+
+it("renders each variation of checked and indeterminate", () => {
+  const variations = [
+    { checked: true, indeterminate: true },
+    { checked: false, indeterminate: true },
+    { checked: true, indeterminate: false },
+    { checked: false, indeterminate: false },
+  ];
+
+  variations.forEach(variation => {
+    expect(
+      renderer.create(<Checkbox label="Foo" {...variation} />).toJSON(),
+    ).toMatchSnapshot();
+  });
+});
+
+describe("Clicking the checkbox it should call the handler", () => {
+  test("with a true value", () => {
+    const clickHandler = jest.fn();
+
+    const { getByLabelText } = render(
+      <Checkbox checked={false} onChange={clickHandler} label="foo" />,
+    );
+
+    fireEvent.click(getByLabelText("foo"));
+    expect(clickHandler).toHaveBeenCalledWith(true);
+  });
+
+  test("with a false value", () => {
+    const clickHandler = jest.fn();
+
+    const { getByLabelText } = render(
+      <Checkbox checked={true} onChange={clickHandler} label="foo" />,
+    );
+
+    fireEvent.click(getByLabelText("foo"));
+    expect(clickHandler).toHaveBeenCalledWith(false);
+  });
+});
```

Comment:
I _think_ this is only relevant on an uncontrolled checkbox. I've expanded the variations snapshot to include `defaultChecked` to ensure that `indeterminate` interacts properly with it as well. Do you think this covers us here?

Reply:
I was more concerned about what happens when you click an indeterminate checkbox. Since it's not exactly clear what the "correct" behaviour here is, it's just a decision we've made it should have a test so that it stays consistent.

Reply:
We talked about this and since the `indeterminate` property is independent of the `checked` property we're not going to automatically change `indeterminate` when `checked` changes.